index.d.ts

```diff
@@ -36,7 +36,7 @@ type PolymorphicExoticComponent<
 		 */
 		<InstanceT extends React.ElementType = T>(
 			props: PolymorphicPropsWithRef<P, InstanceT>,
-		): React.ReactElement | null;
+		): React.ReactNode | null;
```

Comment:
```suggestion
		): ReturnType<React.FunctionComponent>;
```

Found this works better, with different combinations of TS and React types versions, this library wants to always return whatever React itself returns.